File: 프로그래머스/2/42747. H－Index/H－Index.py
from bisect import bisect_left

def calculate_h_index(citations):
    left, right = 0, 10000
    while left <= right:
        mid = (left + right) // 2
        idx = bisect_left(citations, mid)
        count = len(citations) - idx
        
        if count >= mid:
            left += 1
        else:
            right -= 1
    return right

def solution(citations):
    citations = sorted(citations)
    return calculate_h_index(citations)

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    citations.sort((a, b) => b - a)
    let hIndex = 0
    
    for (let i = 0; i < citations.length; i++) {
        if (citations[i] >= i + 1) {
            hIndex = i + 1;
        } else {
            break
        }
    }
    
    return hIndex
}